creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -1034,6 +1082,43 @@ class SnapbackSM {
       this.snapbackReconfigEnabled = this.nodeConfig.get('snapbackReconfigEnabled')
     }
   }
+
+  // Fetch clock values with max number of attempts
+  async fetchClockValues (secondary1, secondary2, wallet, maxClockFetchAttempts = 50) {
+    let secondariesToClockMap = {}
+    let attempts = 0
+
+    while (attempts++ < maxClockFetchAttempts) {
```

Comment:
this loop is pretty confusing to follow, i see what you're doing but might be valuable to clean it up

Reply:
can add comments / adjust per next comment via a discussion